lib/features/composer/domain/usecases/get_autocomplete_with_device_contact_interactor.dart

```diff
@@ -0,0 +1,41 @@
+
+import 'package:core/core.dart';
+import 'package:dartz/dartz.dart';
+import 'package:jmap_dart_client/jmap/mail/email/email_address.dart';
+import 'package:model/model.dart';
+import 'package:tmail_ui_user/features/composer/domain/model/auto_complete_pattern.dart';
+import 'package:tmail_ui_user/features/composer/domain/state/get_autocomplete_state.dart';
+import 'package:tmail_ui_user/features/composer/domain/state/get_device_contact_suggestions_state.dart';
+import 'package:tmail_ui_user/features/composer/domain/usecases/get_autocomplete_interactor.dart';
+import 'package:tmail_ui_user/features/composer/domain/usecases/get_device_contact_suggestions_interactor.dart';
+
+class GetAutoCompleteWithDeviceContactInteractor {
```

Comment:
why not combine some contact sources to one interactor: `SearchEmailAddressInteractor`.